### PR TITLE
[KMSv2] update ci script to create cluster and gather metrics

### PR DESCRIPTION
**Pull request by aramase**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
- Update the KMS CI script to create the cluster and gather metrics

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e/testing-manifests/auth/encrypt/encryption-config.yaml [hunk 1]
```diff
@@ -1,10 +1,71 @@
...
+    - leases.coordination.k8s.io
+    - endpointslices.discovery.k8s.io
+    - events.events.k8s.io
+    - flowschemas.flowcontrol.apiserver.k8s.io
+    - prioritylevelconfigurations.flowcontrol.apiserver.k8s.io
+    - foo2bgmnas.mygroup.example.com
+    - fookb2b2as.mygroup.example.com
+    - foon69n2as.mygroup.example.com
+    - foosc7jjas.mygroup.example.com
+    - foows6b4as.mygroup.example.com
```
Comment (enj) [inline test/e2e/testing-manifests/auth/encrypt/encryption-config.yaml:51]: Uh can we delete these?

Comment (aramase) [reply test/e2e/testing-manifests/auth/encrypt/encryption-config.yaml]: done!

**Comment by aramase** [comment]
/hold cancel

`pull-kubernetes-e2e-kind-kms` is passing with the new changes.

**Pull request merged**